File: source/rolx-client/src/app/core/work-record/record.ts
import { DataWrapper } from '@app/core/util';
import moment from 'moment';
import { DayType } from './day-type';

export interface RecordData {
  date: string;
  dayType: DayType;
  dayName: string;
  nominalWorkTimeHours: number;
}

export class Record extends DataWrapper<RecordData> {

  readonly date = moment(this.raw.date);

  get dayType() {
    return this.raw.dayType;
  }

  get dayName() {
    return this.raw.dayName;
  }

  get nominalWorkTimeHours() {
    return this.raw.nominalWorkTimeHours;
  }

}

File: source/RolXServer/RolXServer/WorkRecord/WebApi/Resource/Record.cs
// -----------------------------------------------------------------------
// <copyright file="Record.cs" company="Christian Ewald">
// Copyright (c) Christian Ewald. All rights reserved.
// Licensed under the MIT license.
// See LICENSE.md in the project root for full license information.
// </copyright>
// -----------------------------------------------------------------------

using System;

namespace RolXServer.WorkRecord.WebApi.Resource
{
    /// <summary>
    /// A record for a day.
    /// </summary>
    public sealed class Record
    {
        /// <summary>
        /// Gets or sets the date.
        /// </summary>
        public DateTime Date { get; set; }

        /// <summary>
        /// Gets or sets the type of the day.
        /// </summary>
        public DayType DayType { get; set; }

        /// <summary>
        /// Gets or sets the name of the day.
        /// </summary>
        public string DayName { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the nominal work-time in hours.
        /// </summary>
        public double NominalWorkTimeHours { get; set; }
    }
}
